### PR TITLE
Video Indexer  Private Endpoint Onboarding

### PR DESCRIPTION
**Pull request by tshaiman**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> 
This PR onboards Azure Video Indexer to have Private Endpoint support for our customers.
It has several new APIs that were added following our integration with the RPaaS team.
The changes were on testing with a preview version from `azure-rest-api-specs-pr` repo for few months
</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/vi.json [hunk 1]
```diff
@@ -0,0 +1,1427 @@
...
+          "example": "Read accounts"
+        },
+        "description": {
+          "description": "Localized friendly description for the operation",
+          "readOnly": true,
+          "type": "string",
+          "example": "Read accounts details"
+        }
+      }
+    },
```
Comment (TimLovellSmith) [inline specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/vi.json:1317]: Are these redundant and removable now you removed the local definition of operationListResult?

Comment (tshaiman) [reply specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/vi.json]: @TimLovellSmith  absolutely ! great catch. all got removed and also moved to using common-types/ErrorDetails

**Comment by tshaiman** [comment]
@TimLovellSmith  @mikekistler The PR Is ready for merge , we only fail on a step that is not related to us 
```
SDK azure-resource-manager-schemas failed
```
it has many javascript error unrelated to us . who is this script owner ? 
can you approve the PR ?

**Comment by mikekistler** [comment]
You can get help on the JavaScript error from the Service API Toolset teams channel. Ping me on Teams if you need a link to that.

**Review by ika-microsoft — APPROVED** [review]

File: specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/examples/ViAccountDelete.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "parameters": {
+    "accountName": "contosto-videoanalyzer",
+    "resourceGroupName": "contoso-rg",
+    "api-version": "2024-06-01-preview",
+    "subscriptionId": "b04775c3-63fa-40f3-9430-139d2e5522d3"
+  },
+  "responses": {
+    "200": {},
```
[the remark this answers is not in the document]
Comment (tshaiman) [reply specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/examples/ViAccountDelete.json:9]: that is part of the requirement of the contract that the Bot are asking to do . TLDR; this is not part of this PR. 
it was there always.

File: specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/examples/ViAccountGet.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      "body": {
+        "id": "/subscriptions/586d4f48-8f08-4a4e-96b7-e1892d6dba9e/resourceGroups/contoso-videoanalyzer-rg/providers/Microsoft.VideoIndexer/accounts/contoso-videoanalyzer",
+        "name": "vi1",
+        "type": "Microsoft.VideoIndexer/accounts",
+        "location": "NorthEurope",
+        "tags": {},
+        "properties": {
+          "tenantId": "8c406f87-77ac-4ebb-a401-e8562450630e",
+          "accountId": "462af7c5-d1f6-4b91-86e3-8bc5e8a61574",
+          "accountName": "contosto-videoanalyzer",
```
[the remark this answers is not in the document]
Comment (tshaiman) [reply specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/examples/ViAccountGet.json]: fixed . but only for this version ( 2024-06-01-preview)  the rest of the previous versions are left with the old name

**Review by ms-bogdan-rotaru — APPROVED** [review]

**Pull request merged**